core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -234,6 +234,11 @@ public static BindableTableScan create(RelOptCluster cluster,
 
     @Override public RelOptCost computeSelfCost(RelOptPlanner planner,
         RelMetadataQuery mq) {
+      boolean noPushing = filters.isEmpty()
+              && projects.size() == table.getRowType().getFieldCount();
+      if (noPushing) {
+        return super.computeSelfCost(planner, mq);
+      }
```

Comment:
Instead of adding this code snippet, wouldn't be sufficient to just remove the factor 0.01d from the multiplier in line 253?

Reply:
No, we should give a preference though, so that the planner would choose BindableTableScan instead of EnumerabeTableScan, without this factor(0.01d), the BindableTableScan would never be choosed if the BindableTableScan has an expression and also pushings.

Reply:
Aren't the factors `f` and `p` sufficient for achieving this?

Reply:
Yes, the `0.01d` factor would ensure that we choose the `BindableTableScan` if we have any pushing projects or filter. The EnumerableInterpreter has another `0.5` overhead of the input table.

Reply:
This is our current choice, the cost estimation is not that accurate and we would rather to give some preference.
For example, why the factor of `EnumerableInterpreter ` is `0.5` ?